contracts/utils/test/lib_rich_errors.ts

```diff
@@ -0,0 +1,34 @@
+import { chaiSetup, constants, provider, txDefaults, web3Wrapper } from '@0x/contracts-test-utils';
+import { BlockchainLifecycle } from '@0x/dev-utils';
+import * as chai from 'chai';
+import * as _ from 'lodash';
+
+import { artifacts, TestLibRichErrorsContract } from '../src';
+
+chaiSetup.configure();
+const expect = chai.expect;
+const blockchainLifecycle = new BlockchainLifecycle(web3Wrapper);
+
+describe('LibEIP712', () => {
+    let lib: TestLibRichErrorsContract;
+
+    before(async () => {
+        await blockchainLifecycle.startAsync();
+        // Deploy SafeMath
+        lib = await TestLibRichErrorsContract.deployFrom0xArtifactAsync(
+            artifacts.TestLibRichErrors,
+            provider,
+            txDefaults,
+        );
+    });
+
+    after(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+
+    describe('_rrevert', () => {
+        it('should correctly revert the extra bytes', async () => {
+            return expect(lib.externalRRevert.callAsync(constants.NULL_BYTES)).to.revertWith(constants.NULL_BYTES);
+        });
+    });
```

Comment:
Not high stakes, but can we get a test with a payload, like `StandardError`? You can use the `@0x/utils.StringRevertError` type (probably my fault that they're named differently):

```ts
// This should work.
const error = new StringRevertError('foo');
return expect(..externalRRevert.callAsync(error.encode()).to.revertWith(error);
```